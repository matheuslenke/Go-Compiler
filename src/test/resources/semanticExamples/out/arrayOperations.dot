digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="import_spec"];
node2[label="Println@1"];
node3[label="parameters"];
node4[label="(string) arg@2"];
node3 -> node4;
node2 -> node3;
node5[label="result"];
node2 -> node5;
node1 -> node2;
node6[label="Scanln@3"];
node7[label="parameters"];
node8[label="(string) arg@4"];
node7 -> node8;
node6 -> node7;
node9[label="result"];
node6 -> node9;
node1 -> node6;
node0 -> node1;
node10[label="(func) testeArrayAssign@5"];
node11[label="parameters"];
node10 -> node11;
node12[label="block"];
node13[label="var_list"];
node14[label="(string) StringVar@6"];
node13 -> node14;
node12 -> node13;
node15[label="var_list"];
node16[label="(int) IntVar@7"];
node15 -> node16;
node12 -> node15;
node17[label="var_list"];
node18[label="(float) Float32Var@8"];
node17 -> node18;
node12 -> node17;
node19[label="var_list"];
node20[label="(float) Float64Var@9"];
node19 -> node20;
node12 -> node19;
node21[label="var_list"];
node22[label="(bool) BoolVar@10"];
node21 -> node22;
node12 -> node21;
node23[label="var_list"];
node24[label="(array) a@11"];
node25[label="array_length"];
node26[label="(int) int@2"];
node25 -> node26;
node24 -> node25;
node27[label="(string) element_type"];
node24 -> node27;
node23 -> node24;
node12 -> node23;
node28[label="assign_list_node"];
node29[label="="];
node30[label="(array) a@11"];
node31[label="(int) index"];
node32[label="(int) int@0"];
node31 -> node32;
node30 -> node31;
node29 -> node30;
node33[label="(string) StringVar@6"];
node29 -> node33;
node28 -> node29;
node12 -> node28;
node34[label="assign_list_node"];
node35[label="="];
node36[label="(array) a@11"];
node37[label="(int) index"];
node38[label="(int) int@1"];
node37 -> node38;
node36 -> node37;
node35 -> node36;
node39[label="(string) string@0"];
node35 -> node39;
node34 -> node35;
node12 -> node34;
node40[label="var_list"];
node41[label="(array) b@12"];
node42[label="array_length"];
node43[label="(int) int@2"];
node42 -> node43;
node41 -> node42;
node44[label="(int) element_type"];
node41 -> node44;
node40 -> node41;
node12 -> node40;
node45[label="assign_list_node"];
node46[label="="];
node47[label="(array) b@12"];
node48[label="(int) index"];
node49[label="(int) int@0"];
node48 -> node49;
node47 -> node48;
node46 -> node47;
node50[label="(int) int@1"];
node46 -> node50;
node45 -> node46;
node12 -> node45;
node51[label="assign_list_node"];
node52[label="="];
node53[label="(array) b@12"];
node54[label="(int) index"];
node55[label="(int) int@1"];
node54 -> node55;
node53 -> node54;
node52 -> node53;
node56[label="(int) int@2"];
node52 -> node56;
node51 -> node52;
node12 -> node51;
node57[label="var_list"];
node58[label="(array) c@13"];
node59[label="array_length"];
node60[label="(int) int@2"];
node59 -> node60;
node58 -> node59;
node61[label="(float) element_type"];
node58 -> node61;
node57 -> node58;
node12 -> node57;
node62[label="assign_list_node"];
node63[label="="];
node64[label="(array) c@13"];
node65[label="(int) index"];
node66[label="(int) int@0"];
node65 -> node66;
node64 -> node65;
node63 -> node64;
node67[label="(float) real@1.5"];
node63 -> node67;
node62 -> node63;
node12 -> node62;
node68[label="assign_list_node"];
node69[label="="];
node70[label="(array) c@13"];
node71[label="(int) index"];
node72[label="(int) int@1"];
node71 -> node72;
node70 -> node71;
node69 -> node70;
node73[label="(float) real@2.5"];
node69 -> node73;
node68 -> node69;
node12 -> node68;
node74[label="var_list"];
node75[label="(array) d@14"];
node76[label="array_length"];
node77[label="(int) int@2"];
node76 -> node77;
node75 -> node76;
node78[label="(float) element_type"];
node75 -> node78;
node74 -> node75;
node12 -> node74;
node79[label="assign_list_node"];
node80[label="="];
node81[label="(array) d@14"];
node82[label="(int) index"];
node83[label="(int) int@0"];
node82 -> node83;
node81 -> node82;
node80 -> node81;
node84[label="(float) real@3.0"];
node80 -> node84;
node79 -> node80;
node12 -> node79;
node85[label="assign_list_node"];
node86[label="="];
node87[label="(array) d@14"];
node88[label="(int) index"];
node89[label="(int) int@1"];
node88 -> node89;
node87 -> node88;
node86 -> node87;
node90[label="(float) real@5.0"];
node86 -> node90;
node85 -> node86;
node12 -> node85;
node91[label="var_list"];
node92[label="(array) e@15"];
node93[label="array_length"];
node94[label="(int) int@2"];
node93 -> node94;
node92 -> node93;
node95[label="(bool) element_type"];
node92 -> node95;
node91 -> node92;
node12 -> node91;
node96[label="assign_list_node"];
node97[label="="];
node98[label="(array) e@15"];
node99[label="(int) index"];
node100[label="(int) int@0"];
node99 -> node100;
node98 -> node99;
node97 -> node98;
node101[label="(bool) bool@true"];
node97 -> node101;
node96 -> node97;
node12 -> node96;
node102[label="assign_list_node"];
node103[label="="];
node104[label="(array) e@15"];
node105[label="(int) index"];
node106[label="(int) int@1"];
node105 -> node106;
node104 -> node105;
node103 -> node104;
node107[label="(bool) bool@false"];
node103 -> node107;
node102 -> node103;
node12 -> node102;
node10 -> node12;
node0 -> node10;
}
