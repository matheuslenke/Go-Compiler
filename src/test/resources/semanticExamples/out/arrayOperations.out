[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:
Entry 0: World




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: testeArrayAssign, line: 5, type: func, scopeName: global
Entry 6 -- name: StringVar, line: 7, type: string, scopeName: testeArrayAssign
Entry 7 -- name: IntVar, line: 8, type: int, scopeName: testeArrayAssign
Entry 8 -- name: Float32Var, line: 9, type: float, scopeName: testeArrayAssign
Entry 9 -- name: Float64Var, line: 10, type: float, scopeName: testeArrayAssign
Entry 10 -- name: BoolVar, line: 11, type: bool, scopeName: testeArrayAssign
Entry 11 -- name: a, line: 13, type: array, subtype: string, scopeName: testeArrayAssign
Entry 12 -- name: b, line: 17, type: array, subtype: int, scopeName: testeArrayAssign
Entry 13 -- name: c, line: 21, type: array, subtype: float, scopeName: testeArrayAssign
Entry 14 -- name: d, line: 25, type: array, subtype: float, scopeName: testeArrayAssign
Entry 15 -- name: e, line: 29, type: array, subtype: bool, scopeName: testeArrayAssign




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.778 s
[INFO] Finished at: 2022-02-24T20:53:40-03:00
[INFO] ------------------------------------------------------------------------
